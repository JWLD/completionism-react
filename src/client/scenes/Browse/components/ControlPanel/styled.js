import styled from 'styled-components'
import FaAlignJustify from 'react-icons/lib/fa/align-justify'
import FaTh from 'react-icons/lib/fa/th'

import { mixins, vars } from 'style'

export const SettingRow = styled.div`
  ${mixins.flexLeft};

  height: ${vars.itemHeight}rem;
  min-height: ${vars.itemHeight}rem;
  width: 100%;
  border-radius: 0.4rem;
  background-color: ${vars.barGrey};
`

export const SettingTitle = styled.span`
  margin-left: 1.5rem;
  margin-right: auto;
  color: white;
  font: 2.5rem DinMC;
`

const SettingIcon = styled.i`
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  padding: 0.75rem;
  padding-top: 0.9rem;
  color: white;

  :hover {
    cursor: pointer;
  }
`

export const ListIcon = styled(SettingIcon.withComponent(FaAlignJustify))`
  color: ${props => (props.active ? 'white' : 'grey')};
`

export const SquaresIcon = styled(SettingIcon.withComponent(FaTh))`
  color: ${props => (props.active ? 'white' : 'grey')};
`
