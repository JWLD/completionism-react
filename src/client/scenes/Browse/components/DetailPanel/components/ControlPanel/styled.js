import styled from 'styled-components'
import FaCheck from 'react-icons/lib/fa/check'

import { mixins, vars } from 'style'

const settingPadding = vars.itemListPadding * 0.75
const checkBoxHeight = 2.2

export const Setting = styled.div`
  ${mixins.flexVertical};

  width: 100%;
  padding: ${settingPadding}rem;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
`

export const SettingTitle = styled.span`
  margin: -0.3rem 0 ${settingPadding * 0.5}rem 0;
  font: 2.25rem DinMC;
  text-transform: uppercase;
`

export const Option = styled.button`
  ${mixins.flex};

  justify-content: space-between;
  width: 100%;
  background-color: transparent;
  color: white;
  font: 2.2rem DinRC;
  line-height: 0;

  :not(:last-child) {
    margin-bottom: ${settingPadding * 0.75}rem;
  }
`

export const CheckBox = styled(FaCheck)`
  height: ${checkBoxHeight}rem;
  width: ${checkBoxHeight}rem;
  padding: 0.2rem;
  border-radius: 0.2rem;
  background-color: ${props => (props.active ? vars.pos : vars.mainGrey)};
  color: ${props => (props.active ? 'white' : vars.mainGrey)};
`
