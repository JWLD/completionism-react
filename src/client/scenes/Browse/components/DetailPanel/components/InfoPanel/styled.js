import styled from 'styled-components'
import FaRocket from 'react-icons/lib/fa/rocket'

import { mixins, vars } from 'style'

export const InfoPanel = styled.div`
  ${mixins.flexVerticalTop};

  width: 100%;
  flex-grow: 1;
  overflow: scroll;

  > *:not(:last-child) {
    margin-bottom: ${vars.itemListPadding}rem;
  }
`

export const ItemName = styled.span`
  ${mixins.flex};

  width: 100%;
  color: white;
  font: 2.5rem DinRC;
  white-space: nowrap;
`

export const ItemImage = styled.div.attrs({
  style: ({ imageUrl }) => ({
    backgroundImage: `url(${imageUrl})`
  })
})`
  width: 100%;
  padding-bottom: 100%;
  border-radius: 0.4rem;
  background-color: #444;
  background-size: cover;
  background-position: center;
`

export const WowheadLink = styled.a`
  ${mixins.flexLeft};

  justify-content: space-between;
  width: 100%;
  height: ${vars.itemHeight}rem;
  padding: 1.2rem;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
  font: 2.5rem DinRC;

  :hover {
    background-color: #555;
  }
`

export const WowheadIcon = styled(FaRocket)`
  height: 100%;
`
