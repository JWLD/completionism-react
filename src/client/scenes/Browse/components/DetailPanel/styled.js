import styled from 'styled-components'
import { FaCog, FaInfoCircle } from 'react-icons/lib/fa'

import { mixins, vars } from 'style'

export const DetailPanel = styled.div`
  ${mixins.flexVerticalTop};

  justify-content: flex-start;
  align-items: flex-start;
  position: sticky;
  top: ${vars.navBarHeight + vars.browsePadding}rem;
  right: ${vars.browsePadding}rem;
  width: 50rem;
  height: calc(100vh - ${vars.navBarHeight + vars.browsePadding * 2}rem);
  padding: ${vars.itemListPadding}rem;
  border-radius: 0.4rem;
  background-color: ${vars.mainGrey};

  > * {
    :not(:last-child) {
      margin-bottom: ${vars.itemListPadding}rem;
    }
  }
`

export const TopBar = styled.div`
  ${mixins.flex};

  height: ${vars.itemHeight}rem;
  min-height: ${vars.itemHeight}rem;
  width: 100%;
`

export const Title = styled.span`
  ${mixins.flex};

  flex-grow: 1;
  height: 100%;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
  font: 3rem DinMC;
  text-transform: uppercase;
`

export const ToggleWrap = styled.button`
  ${mixins.flex};

  position: relative;
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight * 2 - vars.itemHeight * 0.15}rem;
  padding: ${vars.itemHeight * 0.15}rem;
  margin-left: ${vars.itemListPadding}rem;
  border-radius: 0.4rem;
  background-color: ${vars.mainDark};
  overflow: hidden;
`

export const CogIcon = styled(FaCog)`
  position: absolute;
  right: 0;
  top: 0;
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  padding: 0.75rem;
  color: ${props => (props.active ? 'white' : 'gray')};
`

export const InfoIcon = CogIcon.withComponent(FaInfoCircle).extend`
  right: auto;
  left: 0;
`
