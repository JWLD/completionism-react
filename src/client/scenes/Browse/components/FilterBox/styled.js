import styled from 'styled-components'
import { FaTimesCircle } from 'react-icons/lib/fa'

import { mixins, vars } from 'style'

export const FilterInput = styled.input`
  ${mixins.browseItem};

  padding: 1rem;
  background-color: #eee;
  font: 2.5rem DinMC;
`

export const ClearInputBtn = styled(FaTimesCircle)`
  display: ${props => !props.active && 'none'};
  position: absolute;
  right: 0;
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  margin-right: ${vars.itemListPadding}rem;
  padding: 0.75rem;
  color: gray;
  background-color: transparent;
  font-size: 3.25rem;

  :hover {
    cursor: pointer;
  }
`
