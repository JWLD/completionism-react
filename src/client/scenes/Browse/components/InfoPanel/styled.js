import styled from 'styled-components'

import { mixins, vars } from 'style'

export const ItemName = styled.span`
  ${mixins.flex};

  height: ${vars.itemHeight}rem;
  width: 100%;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
  font: 2.5rem DinRC;
  white-space: nowrap;
`

export const ItemImage = styled.div.attrs({
  style: ({ imageUrl }) => ({
    backgroundImage: `url(${imageUrl})`
  })
})`
  width: 100%;
  padding-bottom: 100%;
  border-radius: 0.4rem;
  background-color: #444;
  background-size: cover;
  background-position: center;
`

export const WowheadLink = ItemName.withComponent('a')
