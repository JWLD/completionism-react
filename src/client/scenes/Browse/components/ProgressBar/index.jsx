import React from 'react'
import PropTypes from 'prop-types'

import * as SC from './styled'

const ProgressBar = props => {
  const fillPercent = props.count / props.total * 100

  return (
    <SC.ProgressBarWrap>
      <SC.TextWrap>
        <SC.ProgressText>Temp Title</SC.ProgressText>
        <SC.ProgressText>
          {props.count} / {props.total}
        </SC.ProgressText>
      </SC.TextWrap>

      <SC.ProgressBar>
        <SC.BarFill fill={isNaN(fillPercent) ? 0 : fillPercent} />
      </SC.ProgressBar>
    </SC.ProgressBarWrap>
  )
}

ProgressBar.propTypes = {
  count: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired
}

export default ProgressBar
