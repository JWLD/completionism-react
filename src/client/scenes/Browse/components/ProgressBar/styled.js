import styled from 'styled-components'

import { mixins, vars } from 'style'
import { BrowseBlock } from 'style/components'

const barHeight = 1.5

export const ProgressBarWrap = styled(BrowseBlock)`
  flex-direction: column;
`

export const TextWrap = styled.div`
  ${mixins.flex};

  justify-content: space-between;
  width: 100%;
`

export const ProgressText = styled.span`
  margin: -0.5rem 0;
  font: 3.5rem DinMC;
  color: white;
  text-transform: uppercase;
`

export const ProgressBar = styled.div`
  position: relative;
  overflow: hidden;
  height: ${barHeight}rem;
  width: 100%;
  margin-top: ${vars.itemListPadding}rem;
  background-color: #444;
  border-radius: 0.3rem;
`

export const BarFill = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: ${barHeight}rem;
  width: ${props => props.fill}%;
  background-color: ${vars.pos};
  border-radius: 0.3rem;
`
