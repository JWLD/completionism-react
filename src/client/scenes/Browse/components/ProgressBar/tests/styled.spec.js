import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { BarFill } from 'scenes/Browse/components/ProgressBar/styled'

Enzyme.configure({ adapter: new Adapter() })

const mockProps = {
  fill: 50
}

describe('<BarFill />', () => {
  it('handles [fill] prop correctly', () => {
    const component = shallow(<BarFill {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
