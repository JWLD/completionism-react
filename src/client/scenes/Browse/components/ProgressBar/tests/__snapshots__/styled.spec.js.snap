// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarFill /> handles [fill] prop correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BarFill
    fill={50}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-ifAKCX eBuMpb",
      "fill": 50,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-ifAKCX eBuMpb",
        "fill": 50,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
